grants_tagger/train.py

```diff
@@ -223,14 +224,44 @@ def train_and_evaluate(
             label_binarizer=label_binarizer,
             from_same_distribution=from_same_distribution
         )
-        model.fit(X_train, Y_train)
+        if Y_train.shape[1] > 512:
+            vectorizer = model.steps[0][1]
+            classifier = model.steps[1][1]
+
+            Path(model_path).mkdir(exist_ok=True)
+            print("Fitting vectorizer")
+            vectorizer.fit(X_train)
+            with open(f"{model_path}/tfidf.pkl", "wb") as f:
```

Comment:
this needs to be renamed vectorizer to generalise to keras_vectorizer for example